x-pack/plugins/security_solution/public/flyout/document_details/right/hooks/use_fetch_graph_data.ts

```diff
@@ -61,11 +61,12 @@ export const useFetchGraphData = ({
   req,
   options,
 }: UseFetchGraphDataParams): UseFetchGraphDataResult => {
-  const { actorIds, eventIds, start, end } = req.query;
+  const { eventIds, start, end, esQuery } = req.query;
   const http = useHttp();
+  const QUERY_KEY = ['useFetchGraphData', eventIds, start, end, esQuery];
```

Comment:
should we use `useMemo` here to avoid creating a new array everytime?

Reply:
Hey @PhilippeOberti, thanks for bringing this up! I've been thinking about the use of `useMemo` for the react-query key in this context. 

From what I've gathered, React Query creates a [hash](https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#query-keys-are-hashed-deterministically) of query keys [internally](https://github.com/TanStack/query/blob/f56fc8415904d79a4658167e01a8dbbcf842bd8f/packages/query-core/src/queryClient.ts#L579), in a way that even when creating a new array every time, as long as the contents are the same, React Query will recognize it as the same query and avoid unnecessary refetches.

So I thought that using `useMemo` would not provide a benefit while adding a bit of complexity to the code. But I'm curious, have you noticed any performance improvements or other benefits from memoizing the query key with `useMemo` in your experience? I'd like to hear your thoughts on this!

Reply:
that is a really good point! I think it has become a reflex now when seeing variables like these that I always think about `useMemo` but in this case I agree it is not necessary or even useful at all!